### PR TITLE
Convert AVO SaaS File to Sectors

### PR DESCRIPTION
**Pull request by MrSantamaria**
This PR Updates the AVO olm-artifacts-template.yaml file in order to function with the Sector deployments for progressive delivery. 

We are only applying sectors to the INT stage in our current steps. This PR adds an additional olm-template that is focused on enabling Progressive delivery for HCP enabled regions.

This PR will consume the following SaaS in [app-interface](https://gitlab.cee.redhat.com/service/app-interface/-/merge_requests/69054)

### CODE REVIEW

File: hack/olm-registry/olm-artifacts-template.yaml [hunk 1]
```diff
@@ -168,10 +172,13 @@ objects:
...
       managed.openshift.io/gitRepoName: ${REPO_NAME}
       managed.openshift.io/osd: 'true'
-    name: aws-vpce-operator-hypershift-sss
+    name: aws-vpce-operator-hypershift-sss-${SECTOR}
   spec:
     clusterDeploymentSelector:
       matchExpressions:
+        - api.openshift.com/legal-entity-id: "${OCM_ORG}"
+        - ext-openshift-aws-vpce-operator/cluster: "true"
+        - ext-openshift-aws-vpce-operator/sector: "${SECTOR}"
```
Comment (patjlm) [inline hack/olm-registry/olm-artifacts-template.yaml:181]: I am not aware of the existence of those 2 `ext-openshift-aws-vpce-operator/*` labels. Which component sets them?

Comment (dustman9000) [reply hack/olm-registry/olm-artifacts-template.yaml]: As discussed, let's use the following labels:
```
hive.openshift.io/cluster-region: "${REGION}" <-- this should be available for all ClusterDeployments for all OSD operators
ext-hypershift.openshift.io/cluster-sector: "${SECTOR_NAME}"   <-- this is hypershift specific
```

**Comment by dustman9000** [comment]
This should be coordinated with the needed app-interface changes such that we do not break deployments of AVO.

File: hack/olm-registry/olm-artifacts-template.yaml [hunk 2]
```diff
@@ -168,10 +172,12 @@ objects:
       managed.openshift.io/gitHash: ${IMAGE_TAG}
       managed.openshift.io/gitRepoName: ${REPO_NAME}
       managed.openshift.io/osd: 'true'
-    name: aws-vpce-operator-hypershift-sss
+    name: aws-vpce-operator-hypershift-sss-${SECTOR}
   spec:
     clusterDeploymentSelector:
       matchExpressions:
+        - hive.openshift.io/cluster-region: "${REGION}"
+        - ext-hypershift.openshift.io/cluster-sector: "${SECTOR}"
```
Comment (dustman9000) [inline hack/olm-registry/olm-artifacts-template.yaml:180]:
```suggestion
      - key: ext-hypershift.openshift.io/cluster-region
        operator: In
        values:
        - ${REGION}
```

File: hack/olm-registry/olm-artifacts-template.yaml [hunk 3]
```diff
@@ -168,10 +172,12 @@ objects:
       managed.openshift.io/gitHash: ${IMAGE_TAG}
       managed.openshift.io/gitRepoName: ${REPO_NAME}
       managed.openshift.io/osd: 'true'
-    name: aws-vpce-operator-hypershift-sss
+    name: aws-vpce-operator-hypershift-sss-${SECTOR}
   spec:
     clusterDeploymentSelector:
       matchExpressions:
+        - hive.openshift.io/cluster-region: "${REGION}"
```
Comment (dustman9000) [inline hack/olm-registry/olm-artifacts-template.yaml:179]:
```suggestion
      - key: ext-hypershift.openshift.io/cluster-region
        operator: In
        values:
        - ${REGION}
```

File: hack/olm-registry/olm-artifacts-template.yaml (hunk 2 above)
Comment (dustman9000) [inline hack/olm-registry/olm-artifacts-template.yaml:180]:
```suggestion
        - key: ext-hypershift.openshift.io/cluster-sector: "${SECTOR_NAME}"
           operator: In
           values:
           - ${SECTOR_NAME}
```

File: hack/olm-registry/olm-artifacts-template.yaml [hunk 4]
```diff
@@ -168,10 +172,12 @@ objects:
       managed.openshift.io/gitHash: ${IMAGE_TAG}
       managed.openshift.io/gitRepoName: ${REPO_NAME}
       managed.openshift.io/osd: 'true'
-    name: aws-vpce-operator-hypershift-sss
+    name: aws-vpce-operator-hypershift-sss-${SECTOR}
```
Comment (dustman9000) [inline hack/olm-registry/olm-artifacts-template.yaml:175]:
```suggestion
    name: aws-vpce-operator-hypershift-sss-${SECTOR_NAME}
```

File: hack/olm-registry/olm-artifacts-template.yaml [hunk 5]
```diff
@@ -18,6 +18,10 @@ parameters:
 - name: DISPLAY_NAME
   value: AWS VPCE Operator
   required: true
+- name: SECTOR
```
Comment (dustman9000) [inline hack/olm-registry/olm-artifacts-template.yaml:21]:
```suggestion
- name: SECTOR_NAME
```

**Pull request merged**